src/UglyToad.PdfPig/DocumentLayoutAnalysis/Distances.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Linq;
+using UglyToad.PdfPig.Geometry;
+
+namespace UglyToad.PdfPig.DocumentLayoutAnalysis
+{
+    /// <summary>
+    /// Contains helpful tools for distance measures.
+    /// </summary>
+    public static class Distances
+    {
+        /// <summary>
+        /// The Euclidean distance is the "ordinary" straight-line distance between two points.
+        /// </summary>
+        /// <param name="point1">The first point.</param>
+        /// <param name="point2">The second point.</param>
+        /// <returns></returns>
+        public static double Euclidean(PdfPoint point1, PdfPoint point2)
+        {
+            double dx = (double)(point1.X - point2.X);
+            double dy = (double)(point1.Y - point2.Y);
+            return Math.Sqrt(dx * dx + dy * dy);
+        }
+
+        /// <summary>
+        /// The weighted Euclidean distance.
+        /// </summary>
+        /// <param name="point1">The first point.</param>
+        /// <param name="point2">The second point.</param>
+        /// <param name="wX">The weight of the X coordinates. Default is 1.</param>
+        /// <param name="wY">The weight of the Y coordinates. Default is 1.</param>
+        /// <returns></returns>
+        public static double WghtdEuclidean(PdfPoint point1, PdfPoint point2, double wX = 1.0, double wY = 1.0)
```

Comment:
I think `WeightedEuclidean` would be a better name here, the abbreviation doesn't save enough typing to justify the extra reading effort.